addons/l10n_cn/models/account_move.py

```diff
@@ -13,5 +13,5 @@ class AccountMove(models.Model):
     @api.constrains('fapiao')
     def _check_fapiao(self):
         for record in self:
-            if not record.fapiao.isdecimal():
+            if record.fapiao and (len(record.fapiao) != 8 or not record.fapiao.isdecimal()):
```

Comment:
Ok, should have seen this.  Is the number not required sometimes by the way?  (when it is posted)

Reply:
It's required, but the time you send a fapiao to your customer can vary, it can be before or after the payment. So when you create a invoice, you may don't have a fapiao yet.